pkg/kube/client/ssa/gvk_parser_cache.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ssa
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/crossplane-contrib/provider-kubernetes/apis/v1alpha1"
+)
+
+// GVKParserCacheManager maintains GVK parser cache stores for each provider config.
+type GVKParserCacheManager struct {
+	// mu is used to make sure the cacheStore map is concurrency-safe.
+	mu sync.RWMutex
+	// cacheStore holds the *GVKParserCache per provider configuration.
+	// The cacheStore key is the UID of the provider config object.
+	cacheStore map[types.UID]*GVKParserCache
+}
+
+// GVKParserCacheManagerOption lets you configure a *GVKParserCacheManager.
+type GVKParserCacheManagerOption func(cache *GVKParserCacheManager)
```

Comment:
nit: we don't need these options at the moment, but not against keeping the definition if we need to add in the future

Reply:
let's keep them so that we don't need signature changes in the future